File: frontend/src/DoctorWorkBench/PastHistory.js
import React, { useState, useEffect } from "react";
import "./Navigation.css";
import ToastAlert from '../OtherComponent/ToastContainer/ToastAlert';
import { IconButton } from '@mui/material';
import VisibilityIcon from '@mui/icons-material/Visibility';
import ReactGrid from '../OtherComponent/ReactGrid/ReactGrid';

import { ToastContainer, toast } from "react-toastify";
import { useDispatch,useSelector } from "react-redux";

import axios from 'axios';

function PastHistory() {
  const UrlLink = useSelector(state => state.userRecord?.UrlLink);
  const toast = useSelector(state => state.userRecord?.toast);
  const DoctorWorkbenchNavigation = useSelector(state => state.Frontoffice?.DoctorWorkbenchNavigation);
  const userRecord = useSelector((state) => state.userRecord?.UserData);
  const dispatch = useDispatch();


  

  const [PastHistoryform,setPastHistoryform]=useState({
    illnessordiseases:'No',
    illnessordiseasesText:'',
    surgerybefore:'No',
    surgerybeforeText:'',
    pressureorheartdiseases:'No',
    pressureorheartdiseasesText:'',
    stomachacidityproblem:'No',
    stomachacidityproblemText:'',
    allergicmedicine:'No',
    allergicmedicineText:'',
    drinkalcohol:'No',
    drinkalcoholText:'',
    smoke:'No',
    smokeText:'',
    diabetesorAsthmadisease:'No',
    diabetesorAsthmadiseaseText:'',
    localanesthesiabefore:'No',
    localanesthesiabeforeText:'',
    healthproblems:'No',
    healthproblemsText:'',
    regularbasis:'No',
    regularbasisText:'',
    allergicfood:'No',
    allergicfoodText:'',
    operativeinstuctions:'No',
    operativeinstuctionsText:'',
    Other:'',
  })


 

  const [gridData, setGridData] = useState([]);
  const [IsGetData, setIsGetData] = useState(false)
  const [IsViewMode, setIsViewMode] = useState(false)


  const PastHistorycolumns = [
    { key: 'id', name: 'S.No', frozen: true  },
    { key: 'Type', name: 'Type', frozen: true  },

    // { key: 'VisitId', name: 'VisitId',frozen: true },
    // { key: 'PrimaryDoctorId', name: 'PrimaryDoctorId',frozen: true },
    { key: 'PrimaryDoctorName', name: 'Doctor Name',frozen: true },
    { key: 'Date', name: 'Date',frozen: true },
    { key: 'Time', name: 'Time', frozen: true},
    { key: 'other', name: 'Other' },
    
    {
      key: 'view',
      name: 'View',
      renderCell: (params) => (
        <IconButton onClick={() => handleView(params.row)}>
          <VisibilityIcon />
        </IconButton>
      ),
    },
  ];



  useEffect(() => {
    axios.get(`${UrlLink}Workbench/Workbench_PastHistory_Details`,{params:{RegistrationId:DoctorWorkbenchNavigation?.pk}})
        .then((res) => {
            const ress = res.data
            console.log(ress)
            setGridData(ress)

        })
        .catch((err) => {
            console.log(err);
        })
  }, [UrlLink,DoctorWorkbenchNavigation,IsGetData])


 
  const handelOnchangeform =(e)=>{
    const {name,value}=e.target

    setPastHistoryform((pre)=>({
      ...pre,
      [name]:value,
    }))

  }

 // Handle setting the form data when viewing
const handleView = (data) => {
  setPastHistoryform({
      illnessordiseases: data.illnessordiseases || '',
      illnessordiseasesText: data.illnessordiseasesText || '',
      surgerybefore: data.surgerybefore || '',
      surgerybeforeText: data.surgerybeforeText || '',
      pressureorheartdiseases: data.pressureorheartdiseases || '',
      pressureorheartdiseasesText: data.pressureorheartdiseasesText || '',
      stomachacidityproblem: data.stomachacidityproblem || '',
      stomachacidityproblemText: data.stomachacidityproblemText || '',
      allergicmedicine: data.allergicmedicine || '',
      allergicmedicineText: data.allergicmedicineText || '',
      drinkalcohol: data.drinkalcohol || '',
      drinkalcoholText: data.drinkalcoholText || '',
      smoke: data.smoke || '',
      smokeText: data.smokeText || '',
      diabetesorAsthmadisease: data.diabetesorAsthmadisease || '',
      diabetesorAsthmadiseaseText: data.diabetesorAsthmadiseaseText || '',
      localanesthesiabefore: data.localanesthesiabefore || '',
      localanesthesiabeforeText: data.localanesthesiabeforeText || '',
      healthproblems: data.healthproblems || '',
      healthproblemsText: data.healthproblemsText || '',
      regularbasis: data.regularbasis || '',
      regularbasisText: data.regularbasisText || '',
      allergicfood: data.allergicfood || '',
      allergicfoodText: data.allergicfoodText || '',
      operativeinstuctions: data.operativeinstuctions || '',
      operativeinstuctionsText: data.operativeinstuctionsText || '',
      Other: data.other || '',
  });
  setIsViewMode(true);
};

// Handle clearing the form and resetting the view mode
const handleClear = () => {
  setPastHistoryform({
      illnessordiseases: '',
      illnessordiseasesText: '',
      surgerybefore: '',
      surgerybeforeText: '',
      pressureorheartdiseases: '',
      pressureorheartdiseasesText: '',
      stomachacidityproblem: '',
      stomachacidityproblemText: '',
      allergicmedicine: '',
      allergicmedicineText: '',
      drinkalcohol: '',
      drinkalcoholText: '',
      smoke: '',
      smokeText: '',
      diabetesorAsthmadisease: '',
      diabetesorAsthmadiseaseText: '',
      localanesthesiabefore: '',
      localanesthesiabeforeText: '',
      healthproblems: '',
      healthproblemsText: '',
      regularbasis: '',
      regularbasisText: '',
      allergicfood: '',
      allergicfoodText: '',
      operativeinstuctions: '',
      operativeinstuctionsText: '',
      Other: '',
  });
  setIsViewMode(false);
};



  
  const handleSubmit = () => {
    const dataToSend = {
      RegistrationId:DoctorWorkbenchNavigation?.pk,
      Created_By: userRecord?.username,
      ...PastHistoryform,
      Type:'Doctor',
    };

    console.log(dataToSend,'dataToSend');

    axios.post(`${UrlLink}Workbench/Workbench_PastHistory_Details`, dataToSend)
      .then((res) => {
        const [type, message] = [Object.keys(res.data)[0], Object.values(res.data)[0]];
        dispatch({ type: 'toast', value: { message, type } });
        setIsGetData(prev =>!prev)
        handleClear();
                   
      })
      .catch((err) => console.log(err));

  };


  return (
    <>
   
      <div className="new-patient-registration-form">
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Do you suffer from any illness or diseases ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 1 - cosmetic surgery"
              name="illnessordiseases"
              value={PastHistoryform.illnessordiseases}
              onChange={handelOnchangeform}
              className="new-custom-select-title"
              readOnly={IsViewMode}
              disabled = {IsViewMode}

            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>

           
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="illnessordiseasesText"
              value={PastHistoryform.illnessordiseasesText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.illnessordiseases === "No"}
              readOnly={IsViewMode}
            ></textarea>
            </div>
          </div>
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Have you had any surgery before ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 2 - previous treatments adverse reaction"
              name="surgerybefore"
              value={PastHistoryform.surgerybefore}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              className="new-custom-select-title"
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="surgerybeforeText"
              value={PastHistoryform.surgerybeforeText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.surgerybefore === "No"}
              readOnly={IsViewMode}
            ></textarea>
            </div>
          </div>
        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Did you have high blood pressure or heart diseases ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 3 - allergies to cosmetic products"
              name="pressureorheartdiseases"
              value={PastHistoryform.pressureorheartdiseases}
              onChange={handelOnchangeform}
              className="new-custom-select-title"
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>

            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="pressureorheartdiseasesText"
              value={PastHistoryform.pressureorheartdiseasesText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.pressureorheartdiseases === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>

          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Did you suffer from any stomach acidity problem ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 4 - On medications"
              name="stomachacidityproblem"
              value={PastHistoryform.stomachacidityproblem}
              onChange={handelOnchangeform}
              className="new-custom-select-title"
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="stomachacidityproblemText"
              value={PastHistoryform.stomachacidityproblemText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.stomachacidityproblem === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>
        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Are you allergic to any medicine ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 4 - On medications"
              name="allergicmedicine"
              value={PastHistoryform.allergicmedicine}
              onChange={handelOnchangeform}
              className="new-custom-select-title"
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="allergicmedicineText"
              value={PastHistoryform.allergicmedicineText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.allergicmedicine === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>

          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Do you drink alcohol ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 6 - Smoking Habits"
              className="new-custom-select-title"
              name="drinkalcohol"
              value={PastHistoryform.drinkalcohol}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="drinkalcoholText"
              value={PastHistoryform.drinkalcoholText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.drinkalcohol === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>

        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Do you smoke ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 6 - Smoking Habits"
              className="new-custom-select-title"
              name="smoke"
              value={PastHistoryform.smoke}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
               <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="smokeText"
              value={PastHistoryform.smokeText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.smoke === "No"}
              readOnly={IsViewMode}
            ></textarea>
            </div>
          </div>
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Do you have diabetes/Asthma/any other disease ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 7 - Alcohol Consumptions"
              className="new-custom-select-title"
              name="diabetesorAsthmadisease"
              value={PastHistoryform.diabetesorAsthmadisease}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
               <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="diabetesorAsthmadiseaseText"
              value={PastHistoryform.diabetesorAsthmadiseaseText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.diabetesorAsthmadisease === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>
        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
             Have you had any local anesthesia before ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 8 - Pregnant"
              className="new-custom-select-title"
              name="localanesthesiabefore"
              value={PastHistoryform.localanesthesiabefore}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="localanesthesiabeforeText"
              value={PastHistoryform.localanesthesiabeforeText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.localanesthesiabefore === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>

          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Are you in medication for any health problems?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 9 - breastfeeding"
              className="new-custom-select-title"
              name="healthproblems"
              value={PastHistoryform.healthproblems}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="healthproblemsText"
              value={PastHistoryform.healthproblemsText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.healthproblems === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>
        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Do you take any vitamin supplements or aspirin on regular basis ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 10 - Skin Type"
              className="new-custom-select-title"
              name="regularbasis"
              value={PastHistoryform.regularbasis}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
               <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 5 - Allergies"
              name="regularbasisText"
              value={PastHistoryform.regularbasisText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.regularbasis === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>

          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
             Are you allergic to any food ? <br/>
             please list all types of allergies if any
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 10 - Skin Type"
              className="new-custom-select-title"
              name="allergicfood"
              value={PastHistoryform.allergicfood}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
              <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="allergicfoodText"
              value={PastHistoryform.allergicfoodText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.allergicfood === "No"}
              readOnly={IsViewMode}
            ></textarea>
          </div>
          </div>
        </div>
        <div className="new-patient-info-container">
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Have you received pre and post operative instuctions ?
            </label>
            <div style={{display:'flex',gap:"20px"}}>
            <select
              id="Question 10 - Skin Type"
              className="new-custom-select-title"
              name="operativeinstuctions"
              value={PastHistoryform.operativeinstuctions}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
              disabled = {IsViewMode}
            >
               <option value="">Select</option>
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select>
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="operativeinstuctionsText"
              value={PastHistoryform.operativeinstuctionsText}
              onChange={handelOnchangeform}
              disabled={PastHistoryform.operativeinstuctions === "No"}
              readOnly={IsViewMode}
            ></textarea>
            </div>
          </div>
          <div className="new-custom-form-row width_pasthist">
            <label htmlFor="title" className="new-custom-label-title pasthist">
              Other
            </label>
            <textarea
              className="area_pasthistory"
              id="Question 11 - Other"
              name="Other"
              value={PastHistoryform.Other}
              onChange={handelOnchangeform}
              readOnly={IsViewMode}
            ></textarea>
          </div>
        </div>
        {/* <div className="new-button-container">
          <button className="btncon_add prs-ad-sub-btn" onClick={handleSave}>
            Save
          </button>
        </div> */}
        <div className="Main_container_Btn">
            
          {IsViewMode && (
            <button onClick={handleClear}>Clear</button>
          )}
          {!IsViewMode && (
            <button onClick={handleSubmit}>Submit</button>
          )}
        </div>

        {gridData.length > 0 &&
          <ReactGrid columns={PastHistorycolumns} RowData={gridData} />
        }

        <ToastAlert Message={toast.message} Type={toast.type} />


        {/* <ToastContainer /> */}
      </div>
    </>
  );
}

export default PastHistory;

File: Backend/Workbench/PastHistory.py
from django.http import JsonResponse
from django.views.decorators.csrf import csrf_exempt
from django.views.decorators.http import require_http_methods
from .models import *
import json


@csrf_exempt
@require_http_methods(["POST", "OPTIONS", "GET"])
def Workbench_PastHistory_Details(request):
    if request.method == 'POST':
        try:
            data = json.loads(request.body)
            
            # Extract and validate data
            illnessordiseases = data.get('illnessordiseases', '')
            illnessordiseasesText = data.get('illnessordiseasesText', '')
            surgerybefore = data.get('surgerybefore', '')
            surgerybeforeText = data.get('surgerybeforeText', '')
            pressureorheartdiseases = data.get('pressureorheartdiseases', '')
            pressureorheartdiseasesText = data.get('pressureorheartdiseasesText', '')
            stomachacidityproblem = data.get('stomachacidityproblem', '')
            stomachacidityproblemText = data.get('stomachacidityproblemText', '')
            allergicmedicine = data.get('allergicmedicine', '')
            allergicmedicineText = data.get('allergicmedicineText', '')
            drinkalcohol = data.get('drinkalcohol', '')
            drinkalcoholText = data.get('drinkalcoholText', '')
            smoke = data.get('smoke', '')
            smokeText = data.get('smokeText', '')
            diabetesorAsthmadisease = data.get('diabetesorAsthmadisease', '')
            diabetesorAsthmadiseaseText = data.get('diabetesorAsthmadiseaseText', '')
            localanesthesiabefore = data.get('localanesthesiabefore', '')
            localanesthesiabeforeText = data.get('localanesthesiabeforeText', '')
            healthproblems = data.get('healthproblems', '')
            healthproblemsText = data.get('healthproblemsText', '')
            regularbasis = data.get('regularbasis', '')
            regularbasisText = data.get('regularbasisText', '')
            allergicfood = data.get('allergicfood', '')
            allergicfoodText = data.get('allergicfoodText', '')
            operativeinstuctions = data.get('operativeinstuctions', '')
            operativeinstuctionsText = data.get('operativeinstuctionsText', '')
            other = data.get('Other', '')
            Type = data.get("Type", '')

            
            registration_id = data.get('RegistrationId', '')
            created_by = data.get('created_by', '')
            
            if not registration_id:
                return JsonResponse({'error': 'RegistrationId is required'}, status=400)

            try:
                registration_ins = Patient_Appointment_Registration_Detials.objects.get(pk=registration_id)
            except Patient_Appointment_Registration_Detials.DoesNotExist:
                return JsonResponse({'error': 'Patient not found'}, status=404)

            past_history_instance = Workbench_PastHistory(
                Registration_Id=registration_ins,
                Illnessordiseases=illnessordiseases,
                IllnessordiseasesText=illnessordiseasesText,
                Surgerybefore=surgerybefore,
                SurgerybeforeText=surgerybeforeText,
                Pressureorheartdiseases=pressureorheartdiseases,
                PressureorheartdiseasesText=pressureorheartdiseasesText,
                Stomachacidityproblem=stomachacidityproblem,
                StomachacidityproblemText=stomachacidityproblemText,
                Allergicmedicine=allergicmedicine,
                AllergicmedicineText=allergicmedicineText,
                Drinkalcohol=drinkalcohol,
                DrinkalcoholText=drinkalcoholText,
                Smoke=smoke,
                SmokeText=smokeText,
                DiabetesorAsthmadisease=diabetesorAsthmadisease,
                DiabetesorAsthmadiseaseText=diabetesorAsthmadiseaseText,
                Localanesthesiabefore=localanesthesiabefore,
                LocalanesthesiabeforeText=localanesthesiabeforeText,
                Healthproblems=healthproblems,
                HealthproblemsText=healthproblemsText,
                Regularbasis=regularbasis,
                RegularbasisText=regularbasisText,
                Allergicfood=allergicfood,
                AllergicfoodText=allergicfoodText,
                Operativeinstuctions=operativeinstuctions,
                OperativeinstuctionsText=operativeinstuctionsText,
                Other=other,
                Type=Type,
                created_by=created_by
            )
            past_history_instance.save()
            return JsonResponse({'success': 'PastHistory details added successfully'})
       
        except Exception as e:
            print(f"An error occurred: {str(e)}")
            return JsonResponse({'error': 'An internal server error occurred'}, status=500)
            
    elif request.method == 'GET':
        try:
            registration_id = request.GET.get('RegistrationId')
            if not registration_id:
                return JsonResponse({'warn': 'RegistrationId is required'}, status=400)
            
            try:
                registration_ins = Patient_Appointment_Registration_Detials.objects.get(pk=registration_id)
            except Patient_Appointment_Registration_Detials.DoesNotExist:
                return JsonResponse({'error': 'Patient not found'}, status=404)

            patient_history_datas = Workbench_PastHistory.objects.filter(Registration_Id=registration_ins)

            data = []
            for past_history in patient_history_datas:
                data.append({
                    'id': past_history.Id,
                    'RegistrationId': past_history.Registration_Id.pk,
                    'VisitId': past_history.Registration_Id.VisitId,
                    'PrimaryDoctorId': past_history.Registration_Id.PrimaryDoctor.Doctor_ID,
                    'PrimaryDoctorName': past_history.Registration_Id.PrimaryDoctor.ShortName,
                    'illnessordiseases': past_history.Illnessordiseases,
                    'illnessordiseasesText': past_history.IllnessordiseasesText,
                    'surgerybefore': past_history.Surgerybefore,
                    'surgerybeforeText': past_history.SurgerybeforeText,
                    'pressureorheartdiseases': past_history.Pressureorheartdiseases,
                    'pressureorheartdiseasesText': past_history.PressureorheartdiseasesText,
                    'stomachacidityproblem': past_history.Stomachacidityproblem,
                    'stomachacidityproblemText': past_history.StomachacidityproblemText,
                    'allergicmedicine': past_history.Allergicmedicine,
                    'allergicmedicineText': past_history.AllergicmedicineText,
                    'drinkalcohol': past_history.Drinkalcohol,
                    'drinkalcoholText': past_history.DrinkalcoholText,
                    'smoke': past_history.Smoke,
                    'smokeText': past_history.SmokeText,
                    'diabetesorAsthmadisease': past_history.DiabetesorAsthmadisease,
                    'diabetesorAsthmadiseaseText': past_history.DiabetesorAsthmadiseaseText,
                    'localanesthesiabefore': past_history.Localanesthesiabefore,
                    'localanesthesiabeforeText': past_history.LocalanesthesiabeforeText,
                    'healthproblems': past_history.Healthproblems,
                    'healthproblemsText': past_history.HealthproblemsText,
                    'regularbasis': past_history.Regularbasis,
                    'regularbasisText': past_history.RegularbasisText,
                    'allergicfood': past_history.Allergicfood,
                    'allergicfoodText': past_history.AllergicfoodText,
                    'operativeinstuctions': past_history.Operativeinstuctions,
                    'operativeinstuctionsText': past_history.OperativeinstuctionsText,
                    'other': past_history.Other,
                    'Type': past_history.Type,
                    'created_by': past_history.created_by,
                    'Date': past_history.created_at.strftime('%Y-%m-%d'),  # Format date
                    'Time': past_history.created_at.strftime('%H:%M:%S'),  # Format time
                })

            return JsonResponse(data, safe=False)

        except Exception as e:
            print(f"An error occurred: {str(e)}")
            return JsonResponse({'error': 'An internal server error occurred'},status=500)
    return JsonResponse({'error': 'Method not allowed'}, status=405)
    


